core/server/common/src/main/java/alluxio/master/journal/JournalUtils.java

```diff
@@ -214,5 +220,123 @@ public static void sinkFlush(Supplier<Set<JournalSink>> journalSinks) {
+            || entry.getLastAccessTimeMs() > builder.getLastAccessTimeMs())) {
+          builder.setLastAccessTimeMs(entry.getLastAccessTimeMs());
+        }
+        if (entry.hasMode()) {
+          builder.setMode((short) entry.getMode());
+        }
+        if (entry.getMediumTypeCount() != 0) {
+          builder.clearMediumType();
+          builder.addAllMediumType(new HashSet<>(entry.getMediumTypeList()));
+        }
+        if (entry.hasName()) {
+          builder.setName(entry.getName());
+        }
+        if (entry.hasOwner() && !entry.getOwner().isEmpty()) {
+          builder.setOwner(entry.getOwner());
+        }
+        if (entry.hasParentId()) {
+          builder.setParentId(entry.getParentId());
+        }
+        if (entry.hasPersistenceState()) {
+          builder.setPersistenceState(entry.getPersistenceState());
+        }
+        if (entry.hasPinned()) {
+          builder.setPinned(entry.getPinned());
+        }
+        if (entry.hasTtl()) {
+          builder.setTtl(entry.getTtl());
+        }
+        if (entry.hasTtlAction()) {
+          builder.setTtlAction(entry.getTtlAction());
+        }
+        if (entry.hasUfsFingerprint()) {
+          builder.setUfsFingerprint(entry.getUfsFingerprint());
+        }
+        if (entry.getXAttrCount() > 0) {
+          builder.clearXAttr();
+          builder.putAllXAttr(new HashMap<>(entry.getXAttrMap()));
+        }
+      } else if (oldEntry.hasUpdateInodeFile()) {
+        File.UpdateInodeFileEntry entry = oldEntry.getUpdateInodeFile();
+        File.InodeFileEntry.Builder builder = fileEntryMap.get(entry.getId());
+        if (builder == null) {
+          newEntries.add(oldEntry);
+          continue;
+        }
+        if (entry.hasPersistJobId()) {
+          builder.setPersistJobId(entry.getPersistJobId());
+        }
+        if (entry.hasReplicationMax()) {
+          builder.setReplicationMax(entry.getReplicationMax());
+        }
+        if (entry.hasReplicationMin()) {
+          builder.setReplicationMin(entry.getReplicationMin());
+        }
+        if (entry.hasTempUfsPath()) {
+          builder.setTempUfsPath(entry.getTempUfsPath());
+        }
+        if (entry.hasBlockSizeBytes()) {
+          builder.setBlockSizeBytes(entry.getBlockSizeBytes());
+        }
+        if (entry.hasCacheable()) {
+          builder.setCacheable(entry.getCacheable());
+        }
+        if (entry.hasCompleted()) {
+          builder.setCompleted(entry.getCompleted());
+        }
+        if (entry.hasLength()) {
+          builder.setLength(entry.getLength());
+        }
+        if (entry.getSetBlocksCount() > 0) {
+          builder.clearBlocks();
+          builder.addAllBlocks(entry.getSetBlocksList());
+        }
+      } else {
+        newEntries.add(oldEntry);
+      }
+    }
+    for (File.InodeFileEntry.Builder builder : fileEntryMap.values()) {
+      newEntries.add(Journal.JournalEntry.newBuilder().setInodeFile(builder).build());
```

Comment:
Shall we keep the `InodeFile` entries in the original position in the list?

It is probably not an issue for now but might break natural dependency order if we later use this in other places or add more dependency in inode creation.

Reply:
done